whois-nrtm4/src/main/java/net/ripe/db/nrtm4/SnapshotFileGenerator.java

```diff
@@ -0,0 +1,116 @@
+package net.ripe.db.nrtm4;
+
+import net.ripe.db.nrtm4.persist.NrtmDocumentType;
+import net.ripe.db.nrtm4.persist.NrtmSource;
+import net.ripe.db.nrtm4.persist.NrtmVersionInfo;
+import net.ripe.db.nrtm4.persist.NrtmVersionInfoRepository;
+import net.ripe.db.nrtm4.persist.SnapshotFile;
+import net.ripe.db.nrtm4.persist.SnapshotFileRepository;
+import net.ripe.db.nrtm4.publish.PublishableSnapshotFile;
+import net.ripe.db.nrtm4.publish.SnapshotFileStreamer;
+import net.ripe.db.nrtm4.util.NrtmFileUtil;
+import net.ripe.db.whois.common.dao.SerialDao;
+import net.ripe.db.whois.common.domain.serials.SerialEntry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Optional;
+
+
+@Service
+public class SnapshotFileGenerator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SnapshotFileGenerator.class);
+
+    private final DeltaTransformer deltaTransformer;
+    private final NrtmVersionInfoRepository nrtmVersionInfoRepository;
+    private final SnapshotInitializer snapshotInitializer;
+    private final SnapshotFileStreamer snapshotFileStreamer;
+    private final SnapshotFileRepository snapshotFileRepository;
+    private final SnapshotSynchronizer snapshotSynchronizer;
```

Comment:
Why this class needs SnapshotFileRepository? If both Initializer and synchronizer is already injecting the SnapshotFileRepository and saving to database ?